streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -17,116 +17,109 @@
+     * Create a new instance of {@link KStream} by transforming each element in this stream into a different element in the new stream.
      *
-     * @param mapper the instance of KeyValueMapper
-     * @param <K1>   the key type of the new stream
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link KeyValueMapper}
+     * @param <K1>          the key type of the new stream
+     * @param <V1>          the value type of the new stream
      */
     <K1, V1> KStream<K1, V1> map(KeyValueMapper<K, V, KeyValue<K1, V1>> mapper);
 
     /**
-     * Creates a new instance of KStream by transforming each value in this stream into a different value in the new stream.
+     * Create a new instance of {@link KStream} by transforming each value in this stream into a different value in the new stream.
      *
-     * @param mapper the instance of ValueMapper
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link ValueMapper}
+     * @param <V1>          the value type of the new stream
      */
     <V1> KStream<K, V1> mapValues(ValueMapper<V, V1> mapper);
 
     /**
-     * Creates a new instance of KStream by transforming each element in this stream into zero or more elements in the new stream.
+     * Create a new instance of {@link KStream} by transforming each element in this stream into zero or more elements in the new stream.
      *
-     * @param mapper the instance of KeyValueMapper
-     * @param <K1>   the key type of the new stream
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link KeyValueMapper}
+     * @param <K1>          the key type of the new stream
+     * @param <V1>          the value type of the new stream
      */
     <K1, V1> KStream<K1, V1> flatMap(KeyValueMapper<K, V, Iterable<KeyValue<K1, V1>>> mapper);
 
     /**
-     * Creates a new stream by transforming each value in this stream into zero or more values in the new stream.
+     * Create a new instance of {@link KStream} by transforming each value in this stream into zero or more values in the new stream.
      *
-     * @param processor the instance of Processor
-     * @param <V1>      the value type of the new stream
-     * @return the instance of KStream
+     * @param processor     the instance of {@link ValueMapper}
+     * @param <V1>          the value type of the new stream
      */
     <V1> KStream<K, V1> flatMapValues(ValueMapper<V, Iterable<V1>> processor);
 
     /**
-     * Creates an array of streams from this stream. Each stream in the array corresponds to a predicate in
+     * Create an array of {@link KStream} from this stream. Each stream in the array corresponds to a predicate in
```

Comment:
Oh, and a question just for my understanding: Initially I would have suggest that `branch` should perhaps be named `partition` but then I realized that `branch` is different from (say) Scala's `partition`.  Notably, we ignore/exclude any data records that do not match any of the criteria = no catch-all bucket for `branch`, although this behavior does exist in `partition`.